narwhals/_arrow/series.py

```diff
@@ -769,24 +771,44 @@ def unique(self, *, maintain_order: bool = True) -> Self:
 
     def replace_strict(
         self,
-        old: Sequence[Any] | Mapping[Any, Any],
+        default: Any | NoDefault,
+        old: Sequence[Any],
         new: Sequence[Any],
         *,
         return_dtype: IntoDType | None,
     ) -> Self:
         # https://stackoverflow.com/a/79111029/4451315
         idxs = pc.index_in(self.native, pa.array(old))
+        # Tracks which values were found in the mapping (even if mapped to None)
+        was_matched = pc.is_valid(idxs)
         result_native = pc.take(pa.array(new), idxs)
         if return_dtype is not None:
-            result_native.cast(narwhals_to_native_dtype(return_dtype, self._version))
```

Comment:
😄 did we just cast without returning? 😳 🤦

Reply:
I definitely looks like that 🙈